django/db/models/query.py

```diff
@@ -1249,7 +1245,7 @@ def update(self, **kwargs):
         # Clear any annotations so that they won't be present in subqueries.
         query.annotations = {}
         with transaction.mark_for_rollback_on_error(using=self.db):
-            rows = query.get_compiler(self.db).execute_sql(CURSOR)
+            rows = query.get_compiler(self.db).execute_sql(ROW_COUNT)
```

Comment:
`execute_sql` was ovewritten with `UpdateSQLCompiler` so that `CURSOR` returned a row count, which caused this line to be particularly confusing ("I thought I would get a number back from `update`, but the code returns a cursor?")